### DISCUSSION TITLE
How to implement toggling behaviour for preview (use same keybind to toggle preview command)

### DISCUSSION THREAD

**Discussion by tg0h** [junegunn/fzf#3041] [Q&A] [answered]
<img width="1320" alt="image" src="https://user-images.githubusercontent.com/7954426/200479331-0d05533b-3c57-45ef-bf2a-2bb69d50d625.png">

I am trying to implement a toggle for the preview. If I press ctrl-g once, I intend to display a certain preview, say `cat {}`, If I press ctrl-g again, I want to display another preview, say `bat {}`. I am using a tmp file to store state.

I am unable to get this behaviour. The closest I got was to switch between different keybinds (eg ctrl-g, then ctrl-c, then ctrl-g) in the screenshot above.

**Comment by Ioulan** [comment, marked as answer]
Hello I've tried to do that, but It is not perfect, the way that you wanted it, though it does the job to change the preview.
The problem is:
-  when pressing `ctrl-g` will change the preview into `bat` first, now pressing `ctrl-g` if the preview is on `bat` it will not change the preview So pressing `ctrl-g` sometimes doesn't work and that behavior will be kinda annoying.  ( I don't know why but, I guess the logic or order is wrong I'm too lazy to figure that out xD) here's the code snippet btw:
```sh
fd . | fzf \
  --bind "ctrl-g:execute-silent(touch /tmp/testing; echo "bat" > /tmp/testing)+change-preview(
    if grep -q bat /tmp/testing; then 
      bat --color=always {} && echo "cat" > /tmp/testing;
    else
      cat {} && echo "bat" > /tmp/testing; fi)+refresh-preview"
```

// Don't mind this images xd ( I just uploaded it here to host the image and add this screenshot on the wiki page )
![IMG_20221111_144019](https://user-images.githubusercontent.com/114978689/201280168-550fe07f-a26a-4f8f-8ec9-3c1be05b3316.png)
![1668151301124](https://user-images.githubusercontent.com/114978689/201288486-af070edc-b1c4-4535-bb48-1d51453f501d.png)

**Reply by tg0h** [reply]
thanks @lime-desu  ... I did not know that there was a `+refresh-preview` action

The below is a minimal example of how to toggle a preview with the same keybind
```
#!/usr/bin/env zsh

ls | fzf \
  --bind "ctrl-g:change-preview(
  if [ -f /tmp/testing ]; then 
    cat {}; 
    rm -f /tmp/testing; 
  else 
    bat --color=always {}; 
    touch /tmp/testing; 
  fi
  )+refresh-preview"
```
